### PR TITLE
Update peer-exchange branch with PeerRecord api changes

### PR DESCRIPTION
**Pull request by yusefnapora**
Hey @vyzo, this PR is based on the branch for #234 and just updates it to work with the various name changes that have happened to the signed peer records PR (https://github.com/libp2p/go-libp2p-core/pull/73).

let me know if you have any questions & also feel free to comment on the -core PR if you think we need to change the API at all before merge.

### CODE REVIEW

File: gossipsub.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package pubsub
 import (
 	"context"
 	"fmt"
+	"github.com/libp2p/go-libp2p-core/record"
```
Comment (vyzo) [inline gossipsub.go]: can you put this together with our other imports? Here it is mixed with system imports.

File: gossipsub.go [hunk 2]
```diff
@@ -820,21 +828,24 @@ func (gs *GossipSubRouter) makePrune(p peer.ID, topic string) *pb.ControlPrune {
 		return p != xp
 	})
 
+	cab, peerstoreSupportsSignedAddrs := peerstore.GetCertifiedAddrBook(gs.p.host.Peerstore())
```
Comment (vyzo) [inline gossipsub.go]: can we call the variable something shorter? This is an eyesore.

File: gossipsub.go [hunk 3]
```diff
@@ -304,19 +304,23 @@ func (gs *GossipSubRouter) pxConnect(peers []*pb.PeerInfo) {
...
 				log.Warningf("error unmarshalling routing record obtained through px: %s", err)
 				continue
 			}
-			if srr.PeerID != p {
-				log.Warningf("bogus routing record obtained through px: peer ID %s doesn't match expected peer %s", srr.PeerID, p)
+			rec, ok := r.(*peer.PeerRecord)
+			if !ok {
+				log.Warnf("bogus routing record obtained through px: envelope payload is not PeerRecord")
+			}
+			if rec.PeerID != p {
```
Comment (vyzo) [inline gossipsub.go]: there is a control flow issue here; if `!ok` then `rec.PeerID` is a nil pointer dereference, which could crash.

Comment (yusefnapora) [reply gossipsub.go]: thanks for catching that 👍

**Review by vyzo — APPROVED** [review]

**Pull request merged**